warehouse-management-service/internal/handler/handler.go

```diff
@@ -469,7 +479,226 @@ func (h *handler) DeleteShelfBlock(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+}
+
+func (h *handler) GetShelf(w http.ResponseWriter, r *http.Request) {
+	shelfId := chi.URLParam(r, "shelfId")
+
+	if shelfId == "" {
+		err := fmt.Errorf("%v", api.ShelfResponse{
+			Error: "shelf id cannot be empty",
+		})
+		h.logger.Log(log.Error, err)
+		h.response(w, http.StatusBadRequest, err)
+		return
+	}
 
+	shelf, err := h.shelfService.GetShelfById(r.Context(), shelfId)
+	switch err {
+	case wms.ShelfDoesNotExist:
+		{
+			h.logger.Log(log.Error, err)
+			h.response(w, http.StatusNotFound, api.ShelfResponse{Error: fmt.Sprintf(
+				"failed to get, shelf: %s does not exist",
+				shelfId,
+			)})
+		}
+	case nil:
+		{
+			h.response(w, http.StatusOK, api.ShelfResponse{Response: shelf})
+		}
+	default:
+		{
+			h.logger.Log(log.Error, err)
+			h.response(w, http.StatusInternalServerError, api.ShelfResponse{Error: "Failed to get shelf"})
+		}
+	}
+}
+
+func (h *handler) CreateShelf(w http.ResponseWriter, r *http.Request) {
+	var createShelfRequest api.CreateShelfRequest
+
+	if r.Body == nil {
```

Comment:
This check can be avoided. Error check at 532 should catch this & throw 400?

Reply:
The code on line [529](https://github.com/nilenso/siris-onboarding/pull/34/files/2fccb29993c28887cb5b75d65300dde6193018fb#diff-a3d7cbb31a8120c162b536abd4442e67f77b9a01f0666c4d80ed3f9c8b8f5e3cR529) panics when nil body is passed

Reply:
Got it.

Reply:
Using `json.Decoder` for large volumes of JSON data (in MBs) would be the correct decision.
However, `Decoder` isn't as straightforward as `Unmarshal`.
Checkout the malformed JSON explained in this blog to understand more details: https://dev.to/taqkarim/you-might-not-be-using-json-decoder-correctly-in-golang-12mb

I think we should stick to `Unmarshal` to keep things simple, and go to `Decoder` only when memory becomes a bottleneck for JSON parsing.